### PR TITLE
BadgeNotification

### LINKED ISSUE #3080: Support for BadgeNotification
## What would you like to be added:

Support for numeric badge notifications

## Why is this needed:

UWP feature

## For which Platform:

- [x] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

## Anything else we need to know?

Android and WASM don't have such API

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #3080 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

`BadgeNotification` API not supported

## What is the new behavior?

Numeric `BadgeNotification` supported on iOS and macOS; textual badge supported on macOS.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: doc/articles/features/windows-ui-notifications.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Badge notifications are supported on iOS and macOS.
+
+macOS supports numeric and textual badges (as opposed to UWP) but does not support glyphs.
+
+iOS allows for numeric badges only. It is also necessary to request user permission prior to executing the badge notification for the first time (this request can be executed when appropriate for your application):
+
+```
+#if __IOS__
+UNUserNotificationCenter.Current
```
Comment (jeromelaban) [inline doc/articles/features/windows-ui-notifications.md:13]: Is there a reason why this cannot be included in the BadgeUpdateManager ?

**Comment by MartinZikmund** [comment]
It theoretically could, but then the permission request could appear all of a sudden, out of context (especially if the badge is set in response to some syncing or data update check). This way the developer can show it when appropriate and later update the badge when required. In addition the `Update` and `Clear` methods would finish asynchronously (while the developer would assumed it has already been set).

File: doc/articles/features/windows-ui-notifications.md (hunk 1 above)
Comment (MartinZikmund) [reply doc/articles/features/windows-ui-notifications.md:13]: It theoretically could, but then the permission request could appear all of a sudden, out of context (especially if the badge is set in response to some syncing or data update check). This way the developer can show it when appropriate and later update the badge when required. In addition the Update and Clear methods would finish asynchronously (while the developer would assumed it has already been set).

**Comment by jeromelaban** [comment]
Thanks @MartinZikmund !

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**